File: app/src/main/java/com/example/learn/ui/appbarlayout/AppbarLayoutActivity2.java
package com.example.learn.ui.appbarlayout;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.coordinatorlayout.widget.CoordinatorLayout;
import androidx.viewpager.widget.ViewPager;

import com.example.learn.R;
import com.google.android.material.appbar.AppBarLayout;

public class AppbarLayoutActivity2 extends AppCompatActivity {

    private CoordinatorLayout homeTop;
    private ViewPager homePager;
    private AppBarLayout homeAppBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.appbar_layout_activity2);

        homeTop = (CoordinatorLayout) findViewById(R.id.homeTop);
        homePager = (ViewPager) findViewById(R.id.homePager);
        homeAppBar = (AppBarLayout) findViewById(R.id.homeAppBar);

        homePager.setAdapter(new AppbarLayoutPagerAdapter(getSupportFragmentManager()));

    }
}

File: app/src/main/java/com/example/learn/ui/appbarlayout/AppbarLayoutActivity2.kt
package com.example.learn.ui.appbarlayout

import android.os.Bundle
import com.example.common.base.BaseActivity
import com.example.learn.R
import kotlinx.android.synthetic.main.appbar_layout_activity2.*

class AppbarLayoutActivity2 : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.appbar_layout_activity2)
        homePager.adapter = AppbarLayoutPagerAdapter(supportFragmentManager)
    }

}